### ISSUE TITLE
ein-org.el dramatically increases org-capture startup time

### ISSUE THREAD

**Issue by leezu** [millejoh/emacs-ipython-notebook#265]
When calling org-capture for the first time in an emacs session, without previously interacting with ipython notebooks / ein, `ein-org.el` increases the `org-capture` startup time from instant to ~10 seconds. This shouldn't be the case.

**Comment by leezu** [comment]
Profiling graph looks as follows:

```
- command-execute                                         445,354,393  99%
 - call-interactively                                     443,900,975  99%
  - funcall-interactively                                 431,497,706  96%
   - org-capture                                          420,842,575  94%
    - org-store-link                                      416,547,366  93%
     - ein:org-store-link                                 399,610,662  89%
      - byte-code                                         399,558,144  89%
       - require                                          399,116,308  89%
        - byte-code                                       399,030,200  89%
         - require                                        396,096,727  88%
          - byte-code                                     395,880,483  88%
           - ein:generate-menu                            249,483,262  55%
            - mapcar                                      249,483,262  55%
             + #<compiled 0x2a8d155>                      249,483,262  55%
           - require                                      145,588,369  32%
            - byte-code                                   118,509,142  26%
             - require                                    118,005,526  26%
              - byte-code                                 117,809,722  26%
               - ein:generate-menu                        112,421,028  25%
                - mapcar                                  112,421,028  25%
                 + #<compiled 0x2a8d155>                  112,418,916  25%
               + require                                    1,340,438   0%
               + custom-declare-variable                        3,168   0%
              + #<compiled 0x2d228fd>                         147,936   0%
                do-after-load-evaluation                       26,648   0%
              + file-truename                                  21,180   0%
            + do-after-load-evaluation                     27,021,057   6%
            + file-truename                                    27,998   0%
            + #<compiled 0x2d228fd>                            26,036   0%
              defvar                                            4,136   0%
           + easy-menu-do-define                               69,304   0%
          + #<compiled 0x2d228fd>                              97,580   0%
          + do-after-load-evaluation                           89,420   0%
          + file-truename                                      26,140   0%
          + substitute-env-in-file-name                         2,048   0%
         + ein:generate-menu                                1,849,445   0%
        + #<compiled 0x2d228fd>                                43,872   0%
          do-after-load-evaluation                             26,648   0%
        + file-truename                                         7,400   0%
       + eval-after-load                                        9,216   0%
        do-after-load-evaluation                               25,624   0%
      + file-truename                                          14,522   0%
      + #<compiled 0x2d228fd>                                  10,300   0%
      + substitute-env-in-file-name                             1,024   0%
     + org-load-modules-maybe                              16,672,168   3%
    + org-capture-select-template                           4,295,209   0%
   + helm-M-x                                              10,655,131   2%
  + progn                                                  12,403,269   2%
 + autoload-do-load                                         1,024,522   0%

```

**Comment by millejoh** [comment]
Thank you. The ein codebase is relatively big and it is difficult to load one part without loading everything else. I'm sure there is a better way, but I'll need a true emacs expert to point the way. In the meantime I can think of a couple solutions:

1) Ensure all of ein is loaded when Emacs first starts up. You'll still have a long delay as ein loads, but at least you will get it out of the way before starting any real work.

2) ein-org has an `eval-after-load` form that executes after org is loaded. This could be changed to something else, or maybe eliminated altogether and some other mechanism put in place to detect when both ein and org have been loaded.

**Comment by braham-snyder** [comment]
Removing the `:store` kwarg from the `org-link-set-parameters` call in `ein-autoloads` suffices (but is obviously not a good solution) -- perhaps that `:store` kwarg should be set only after `ein` (or `ein-notebook`?) has loaded.

**Comment by millejoh** [comment]
I've seen vast improvements in loading performance for ein. Given the current lack of conversation I am going to close this issue, but if you still are seeing performance hits feel free to reopen or submit a new issue. Thanks.
